sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityCredential.java

```diff
@@ -53,8 +56,16 @@ public String getClientId() {
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        return (appServiceMSICredential != null
-            ? appServiceMSICredential.authenticate(request)
-            : virtualMachineMSICredential.authenticate(request));
+        Mono<AccessToken> accessTokenMono;
+        if (appServiceMSICredential != null) {
+            accessTokenMono = appServiceMSICredential.authenticate(request);
+            logger.info("Azure Identity => Managed Identity environment: MSI_ENDPOINT");
+        } else {
+            accessTokenMono = virtualMachineMSICredential.authenticate(request);
+            logger.info("Azure Identity => Managed Identity environment: IMDS");
```

Comment:
This can be misleading in cases when running on non-Azure hosted machines.
As we default to VM credential if MSI endpoint is not set.

Reply:
I've moved this log to be only printed in doOnSuccess().